compiler/src/test/java/com/squareup/anvil/compiler/codegen/ContributesMultibindingGeneratorTest.kt

```diff
@@ -325,32 +353,47 @@ class ContributesMultibindingGeneratorTest {
 
       @ContributesMultibinding(Int::class, boundType = Any::class)
       interface ContributingInterface
-      """
+      """,
+      mode = mode,
     ) {
       assertThat(contributingInterface.hintMultibinding?.java).isEqualTo(contributingInterface)
       assertThat(contributingInterface.hintMultibindingScope).isEqualTo(Int::class)
     }
   }
 
   @Test fun `a contributed multibinding can be generated`() {
-    val codeGenerator = simpleCodeGenerator { clazz ->
-      clazz
-        .takeIf { it.isAnnotatedWith(mergeComponentFqName) }
-        ?.let {
-          //language=kotlin
-          """
-            package com.squareup.test
-                
-            import com.squareup.anvil.annotations.ContributesMultibinding
-            import dagger.MapKey
-            import javax.inject.Singleton
-
-            @ContributesMultibinding(Any::class)
-            @BindingKey("abc")
-            @Singleton
-            interface ContributingInterface : ParentInterface
-          """.trimIndent()
+    val stubContentToGenerate =
+      //language=kotlin
+      """
+          package com.squareup.test
+              
+          import com.squareup.anvil.annotations.ContributesMultibinding
+          import dagger.MapKey
+          import javax.inject.Singleton
+       
+          @ContributesMultibinding(Any::class)
+          @BindingKey("abc")
+          @Singleton
+          interface ContributingInterface : ParentInterface
+      """.trimIndent()
+
+    val localMode = when (mode) {
+      is AnvilCompilationMode.Embedded -> {
+        val codeGenerator = simpleCodeGenerator { clazz ->
+          clazz
+            .takeIf { it.isAnnotatedWith(mergeComponentFqName) }
+            ?.let { stubContentToGenerate }
+        }
+        AnvilCompilationMode.Embedded(listOf(codeGenerator))
+      }
+      is AnvilCompilationMode.Ksp -> {
+        val processor = simpleSymbolProcessor { resolver ->
+          resolver.getSymbolsWithAnnotation(MergeComponent::class.qualifiedName!!)
```

Comment:
Is there any reason to use `qualifiedName` over `canonicalName` or vice versa? Just asking since I noticed you used `getSymbolsWithAnnotation(ContributesMultibinding::class.java.canonicalName)` in the code gen

Reply:
No functional difference, one is just java reflection and the other is kotlin reflection. Thinking we could maybe standardize toward the latter to make it more KMP-friendly in the future?

Reply:
👍 Sounds good to me